pyuvdata/uvcal/uvcal.py

```diff
@@ -555,6 +621,173 @@ def metadata_only(self):
+                    getattr(self, "_" + param_name).form = (
+                        "Nfreqs",
+                        "Ntimes",
+                        "Njones",
+                    )
+
+    def use_future_array_shapes(self):
+        """
+        Change the array shapes of this object to match the planned future shapes.
+
+        This method sets allows users to convert to the planned array shapes changes
+        before the changes go into effect. This method sets the `future_array_shapes`
+        parameter on this object to True.
+
+        """
+        self._set_future_array_shapes()
+        if not self.metadata_only:
+            # remove the length-1 spw axis for all data-like parameters
+            for param_name in self._data_params:
+                param_value = getattr(self, param_name)
+                if param_value is None:
+                    continue
+                if param_name == "total_quality_array":
+                    setattr(self, param_name, (param_value)[0, :, :, :])
+                else:
+                    setattr(self, param_name, (param_value)[:, 0, :, :])
+
+        # remove the length-1 spw axis for the freq_array
+        self.freq_array = self.freq_array[0, :]
+
+        # force integration_time to be an array of length Ntimes
+        self.integration_time = (
+            np.zeros(self.Ntimes, dtype=np.float64) + self.integration_time
+        )
+
+        if not self.flex_spw:
+            # make channel_width be an array of length Nfreqs rather than a single value
+            # (not needed with flexible spws because this is already done in that case)
+            self.channel_width = (
+                np.zeros(self.Nfreqs, dtype=np.float64) + self.channel_width
+            )
+
+    def use_current_array_shapes(self):
+        """
+        Change the array shapes of this object to match the current future shapes.
+
+        This method sets allows users to convert back to the current array shapes.
+        This method sets the `future_array_shapes` parameter on this object to False.
+        """
+        if not self.flex_spw:
+            unique_channel_widths = np.unique(self.channel_width)
+            if unique_channel_widths.size > 1:
+                raise ValueError(
+                    "channel_width parameter contains multiple unique values, but "
+                    "only one spectral window is present. Cannot collapse "
+                    "channel_width to a single value."
+                )
+            self._channel_width.form = ()
+            self.channel_width = unique_channel_widths[0]
+
+            unique_integration_times = np.unique(self.integration_time)
+            if unique_integration_times.size > 1:
+                raise ValueError(
+                    "integration_time parameter contains multiple unique values. "
+                    "Cannot collapse integration_time to a single value."
+                )
+            self._integration_time.form = ()
```

Comment:
This doesn't seem like it would depend on the `flex_spw` attribute (and I'm not actually certain it depends on current/future array shapes, given the changes in the parameter attributes above).

Reply:
This is fixed, it only depends on using future_array_shapes.